core/src/services/hdfs/backend.rs

```diff
@@ -155,6 +155,34 @@ pub struct HdfsBackend {
 unsafe impl Send for HdfsBackend {}
 unsafe impl Sync for HdfsBackend {}
 
+impl HdfsBackend {
+    fn create_parent_if_need(&self, path: &str) -> Result<()> {
```

Comment:
Please use 

```rust
self.client
    .create_dir(&parent.to_string_lossy())
    .map_err(new_std_io_error)?;
```

like what we do in other methods.

Reply:
Do you mean the method name?

Reply:
> Do you mean the method name?

I feel like we don't need this new function. How using `self.client.create_dir` instead?

Reply:
removed